.github/workflows/build_and_test.yml

```diff
@@ -166,15 +168,21 @@ jobs:
             exit 1
           fi
           echo "CLIPPY_COMMON_ARGS=${CLIPPY_COMMON_ARGS}" >> $GITHUB_ENV
+
       - name: Run cargo clippy (debug)
-        run: cargo hack --feature-powerset clippy $CLIPPY_COMMON_ARGS
-      - name: Run cargo clippy (release)
-        run: cargo hack --feature-powerset clippy --release $CLIPPY_COMMON_ARGS
+        run: |
+          parallel --jobs 8 "cargo hack --feature-powerset --partition {}/8 clippy --target-dir target/partition-{} $CLIPPY_COMMON_ARGS" ::: 1 2 3 4 5 6 7 8
+
+      # instead of running the full release build, running debug build again,
+      # but with disabled `debug-assertions` to excersice release code paths
+      - name: Run cargo clippy (debug, with debug-assertions=false)
```

Comment:
Reading through https://doc.rust-lang.org/cargo/reference/profiles.html , it appears that `debug-assertions` is indeed the only `cfg!()`-detectable difference between `cargo clippy` and `cargo clippy --release`. 

I don't know if this is best practice in the ecosystem though, prob @arpad-m or @koivunej would know better.

Reply:
debug-assertions don't bring anything for check only checks, so I think it's okay to disable them.